src/index.js

```diff
@@ -39,7 +39,14 @@ if (
 }
 
 /* Warning if there are several instances of styled-components */
-if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
+if (
+  process.env.NODE_ENV !== 'production' &&
+  process.env.NODE_ENV !== 'test' &&
+  typeof window !== 'undefined' &&
+  typeof navigator !== 'undefined' &&
```

Comment:
Can we make sure to also check `typeof navigator.userAgent === 'string'` just to be sure. These "faux" envs can be quite unstable